plugins/modules/ec2_imagebuilder_component.py

```diff
@@ -0,0 +1,399 @@
+    tags:
+      this: isatag
+
+- name: publish a new version of a component
+  ec2_imagebuilder_component:
+    name: test_component
+    data:
+      schemaVersion: "1.0"
+      phases:
+        - name: "build"
+          steps:
+            - name: "RunScript"
+              action: "ExecuteBash"
+              timeoutSeconds: 120
+              onFailure: "Abort"
+              maxAttempts: 3
+              inputs:
+                commands:
+                  - |
+                    #!/bin/bash
+                    echo 'new and improved!'
+                    ...
+    description: this is a test component -- updated
+    platform: Linux
+    semantic_version: 1.0.1
+    state: present
+    tags:
+      this: isatag
+
+- name: update tags for existing component version
+  ec2_imagebuilder_component:
+    name: test_component
+    platform: Linux
+    semantic_version: 1.0.1
+    state: present
+    tags:
+      this: isatag
+      another: tag
+
+- name: delete a version of a component
+  ec2_imagebuilder_component:
+    name: test_component
+    semantic_version: 1.0.0
+    state: absent
+'''
+
+RETURN = '''
+component:
+  description: Component configuration details. \
+    Refer to U(https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/imagebuilder.html#imagebuilder.Client.get_component)
+  returned: always
+  type: complex
+  contains:
+    arn:
+      description: The ARN of the resource
+      returned: always
+      type: str
+    change_description:
+      returned: when state is present
+      type: str
+      description: ''
+    data:
+      returned: when state is present
+      type: str
+      description: ''
+    date_created:
+      description: The time and date that this component version was created.
+      returned: when state is present
+      type: str
+      sample: "2018-04-21T05:19:58.326000+00:00"
+    description:
+      returned: when state is present
+      type: str
+      description: ''
+    encrypted:
+      returned: when state is present
+      type: bool
+      description: ''
+    uri:
+      returned: when state is present
+      type: str
+      description: ''
+    kms_key_id:
+      returned: when state is present
+      type: str
+      description: ''
+    name:
+      returned: when state is present
+      type: str
+      description: ''
+    owner:
+      returned: when state is present
+      type: str
+      description: ''
+    platform:
+      returned: when state is present
+      type: str
+      description: ''
+    semantic_version:
+      returned: when state is present
+      type: str
+      description: ''
+    tags:
+      returned: when state is present
+      type: dict
+      description: ''
+    type:
+      returned: when state is present
+      type: str
+      description: ''
+'''
+
+import copy
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import get_aws_connection_info, camel_dict_to_snake_dict, snake_dict_to_camel_dict
```

Comment:
camel_dict_to_snake_dict should actually come from ansible.module_utils.common.dict_transformations now

```suggestion
from ansible_collections.amazon.aws.plugins.module_utils.ec2 import get_aws_connection_info
from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict, snake_dict_to_camel_dict
```